iceoryx_binding_c/source/c_listener.cpp

```diff
@@ -57,6 +57,25 @@ ENUM iox_ListenerResult iox_listener_attach_subscriber_event(iox_listener_t cons
     return ListenerResult_SUCCESS;
 }
 
+ENUM iox_ListenerResult
+iox_listener_attach_subscriber_event_with_context_data(iox_listener_t const self,
+                                                       iox_sub_t const subscriber,
+                                                       const ENUM iox_SubscriberEvent subscriberEvent,
+                                                       void (*callback)(iox_sub_t, void*),
+                                                       void* const contextData)
+{
+    EventCallback<cpp2c_Subscriber, void> eventCallback;
+    eventCallback.m_callback = callback;
+    eventCallback.m_contextData = contextData;
```

Comment:
So this is where context comes from, in the C++ class storing it is called the same. So for consistency it makes sense (but I consider it a misnomer still, at all levels).